File: eternal_return/src/main/java/org/EternalReturn/ERCharacter/Character/lidailin/Character_LiDailin.java
package org.EternalReturn.ERCharacter.Character.lidailin;

import org.EternalReturn.ERCharacter.ERCharacter;
import org.EternalReturn.ERPlayer.ERPlayer;

public class Character_LiDailin extends ERCharacter {

    public boolean isDrunk = false;
    public int drinkCount = 0;

    public Character_LiDailin(ERPlayer player) {
        super(player);
        this.cooldownSeconds = 8;
        registerMonobehaviour(new Passive());
        registerMonobehaviour(new Active());
    }


    @Override
    public String getName() { return "lidailin"; }

}

File: eternal_return/src/main/kotlin/org/eternalreturn/ercharacter/character/lidailin/Character_LiDailin.kt
package org.eternalreturn.ercharacter.character.lidailin

import org.bukkit.entity.Player
import org.eternalreturn.ercharacter.datastructure.CoolTableSeconds
import org.eternalreturn.erplayer.DPlayer
import org.eternalreturn.system.EREngine

class Character_LiDailin(erEngine : EREngine, player: Player) : DPlayer(player, erEngine) {
    var isDrunk: Boolean = false
    var drinkCount: Int = 0
    override val activeCoolForEachLevel  = CoolTableSeconds(this::activeLevel, 25, 24, 23, 22, 20)
    override val passiveCoolForEachLevel = CoolTableSeconds(this::activeLevel, 10, 10, 10, 10, 10)

    init {
        this.ActiveCooldownSeconds = 3
        this.PassiveCooldownSeconds = 8
        registerMonobehaviour(Passive())
        registerMonobehaviour(Active())
        registerMonobehaviour(DrunkTimer());
    }


    override val name: String
        get() = "lidailin"
}